runtime/src/main/cpp/Exceptions.cpp

```diff
@@ -227,79 +226,110 @@ void OnUnhandledException(KRef throwable) {
   }
 }
 
-#if KONAN_REPORT_BACKTRACE_TO_IOS_CRASH_LOG
-static bool terminating = false;
-static SimpleMutex terminatingMutex;
-#endif
+namespace {
 
-RUNTIME_NORETURN void TerminateWithUnhandledException(KRef throwable) {
-  OnUnhandledException(throwable);
+class {
+    /**
+     * Timeout 5 sec for concurrent (second) terminate attempt to give a chance the first one to finish.
+     * If the terminate handler hangs for 5 sec it is probably fatally broken, so let's do abnormal _Exit in that case.
+     */
+    unsigned int timeoutSec = 5;
```

Comment:
Now that I thought about the test, I'm starting to wonder: (1) should this value be used to limit termination handling regardless whether there's a concurrent request or not? (2) if so, should this be configurable, so that if someone's unhandled exception hook is really slow, they can manually tweak the value.

It doesn't have to be resolved in this PR.

Reply:
W.r.t. configurable timeout: agree, it may be useful for debug purpose or in case of slow target. Let me consider this in the scope of stress-test related commit (TBD)